flanker/mime/message/headers/headers.py

```diff
@@ -34,8 +34,12 @@ def __contains__(self, key):
         return normalize(key) in self._v
 
     def __setitem__(self, key, value):
-        self._v[normalize(key)] = remove_newlines(value)
-        self.changed = True
+        key = normalize(key)
+        if key in self._v:
+            self._v[key] = remove_newlines(value)
+            self.changed = True
+        else:
+            self.prepend(key, remove_newlines(value))
```

Comment:
prepend handles that. prepend records that this field was prepended. If no other changes are made, we will serialize just the prepended fields and then output the original message unchanged.